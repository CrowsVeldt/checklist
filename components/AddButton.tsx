import { router } from "expo-router";
import { Pressable, StyleSheet, Text } from "react-native";

export default function AddButton() {
  return (
    <Pressable
      style={({ pressed }) =>
        pressed ? [styles.button, styles.buttonPressed] : styles.button
      }
      onPress={() => router.push("/create")}
    >
      <Text style={styles.buttonText}>+</Text>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    height: 50,
    width: 50,
    borderColor: "black",
    borderWidth: 1,
    borderRadius: 5,
    position: "absolute",
    bottom: 20,
    right: 20,
  },
  buttonPressed: {
    backgroundColor: "gray",
  },
  buttonText: {
    height: "100%",
    width: "100%",
    textAlign: "center",
    textAlignVertical: "center",
  },
});
