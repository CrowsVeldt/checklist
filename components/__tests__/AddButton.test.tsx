import { render } from "@testing-library/react-native";
import AddButton from "../AddButton";

describe("<AddButton />", () => {
  test("AddButton renders correctly", () => {
    const tree = render(<AddButton />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
