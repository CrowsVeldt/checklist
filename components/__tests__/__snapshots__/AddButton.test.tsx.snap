// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddButton /> AddButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "black",
      "borderRadius": 5,
      "borderWidth": 1,
      "bottom": 20,
      "height": 50,
      "justifyContent": "center",
      "position": "absolute",
      "right": 20,
      "width": 50,
    }
  }
>
  <Text
    style={
      {
        "height": "100%",
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": "100%",
      }
    }
  >
    +
  </Text>
</View>
`;
